python/ambassador/envoy/v2/v2listener.py

```diff
@@ -562,7 +562,7 @@ def finalize_tcp(self) -> None:
                 # First up, which clusters do we need to talk to?
                 clusters = [{
                     'name': mapping.cluster.envoy_name,
-                    'weight': mapping.weight
+                    'weight': mapping._weight or 0  # It's not actually possible for _weight to be unset
```

Comment:
... so why is it `Optional[int]` and why does this `or 0` need to be here?

Reply:
Is it saying that it's not possible for it to be unset at this point in the lifecycle? And the `or 0` is just to shut up mypy?

If so, I'd prefer to instead on line 562ish, stick a `assert(isinstance(mapping._weight, int))`. This'll satisfy mypy, check that belief at low `python -O` levels, and optimize that check out at high `python -O` levels.

Reply:
I had a long rationale for this typed up, but realized that really, the long rationale can be avoided and we can just set it to 0 at instantiation and that's not worse than what's happening right now. 🤷‍♂️